lib/game.rb

```diff
@@ -31,123 +35,102 @@ def self.count_of_games_by_season
+      end
+      hash_divided
     end
-  end
 
-  def self.total_goals_per(csv_header)
-    goals_per(csv_header, :away_goals) + goals_per(csv_header, :home_goals)
+  def self.games_goals_by_season
+    hash_of_hashes(all, :season, :goals, :games_played, :total_goals, 1)
   end
 
-  def self.average_goals(sum_array, length_array)
-    sum_array.each_with_index.map do |goals, index|
-      (goals.to_f / length_array[index].to_f ).round(2)
+  #deliverable
+    def self.average_goals_by_season
+      # :goals / :games_played
+      divide_hash_values(:goals, :games_played, games_goals_by_season)
+    end
+
+  def self.games_goals_by(hoa_team)
+    #{away_team_id => {goals => x, games_played => y}}
+    if hoa_team == :away_team
+      hash_of_hashes(all, :away_team_id, :goals, :games_played, :away_goals, 1)
+    elsif hoa_team == :home_team
+      hash_of_hashes(all, :home_team_id, :goals, :games_played, :home_goals, 1)
     end
   end
 #deliverable
-  def self.average_goals_by_season
-    avg_goals_per_season = average_goals(total_goals_per(:season), games_per(:season))
-    season_ids = @@all.map { |game| game.season}.uniq
-    # create a hash of {season_ids => average_goals_per_season}
-    Hash[season_ids.zip(avg_goals_per_season)]
+  def self.average_goals_by(hoa_team)
+      divide_hash_values(:goals, :games_played, games_goals_by(hoa_team))
   end
-
-#deliverable (needs to access teams.csv via stat_tracker)
+#deliverable
   def self.highest_scoring_visitor_team_id
-    avg_away_goals = average_goals(goals_per(:away_team_id, :away_goals), games_per(:away_team_id))
-    away_team_ids = @@all.map { |game| game.away_team_id }.uniq
-    # create hash of {team_ids => average goals}
-    away_ids_n_goals = Hash[away_team_ids.zip(avg_away_goals)]
-    away_ids_n_goals.max_by{ |team_id, away_goals| away_goals}.first
+    average_goals_by(:away_team).max_by{ |team_id, away_goals| away_goals}.first
   end
-#deliverable (needs to access teams.csv via stat_tracker)
+#deliverable
   def self.highest_scoring_home_team_id
-    avg_home_goals = average_goals(goals_per(:home_team_id, :home_goals), games_per(:home_team_id))
-    home_team_ids = @@all.map { |game| game.home_team_id }.uniq
-    # create hash of {team_ids => average goals}
-    home_ids_n_goals = Hash[home_team_ids.zip(avg_home_goals)]
-    home_ids_n_goals.max_by{ |team_id, home_goals| home_goals}.first
+    average_goals_by(:home_team).max_by{ |team_id, away_goals| away_goals}.first
   end
-#deliverable (needs to access teams.csv via stat_tracker)
+#deliverable
   def self.lowest_scoring_visitor_team_id
-    avg_away_goals = average_goals(goals_per(:away_team_id, :away_goals), games_per(:away_team_id))
-    away_team_ids = @@all.map { |game| game.away_team_id }.uniq
-    # create hash of {team_ids => average goals}
-    away_ids_n_goals = Hash[away_team_ids.zip(avg_away_goals)]
-    away_ids_n_goals.min_by{ |team_id, away_goals| away_goals}.first
```

Comment:
LOOK AT ALL THIS DELETED CODE!

Reply:
![](https://media0.giphy.com/media/kqu0JnieiuhGw/giphy.gif?cid=5a38a5a2907f5f684c3302f6e9e4fb885123258e26a65bef&rid=giphy.gif)

Reply:
Nice!!!